src/game/memory.c

```
@@ -185,10 +185,42 @@ void growing_pool_free_pool(struct GrowingPool *pool) {
+            // if realloc fails, destroy the array and create a new one
+            if (!buffer) {
+                growing_array_free(&array);
+                return growing_array_init(NULL, capacity, alloc, free);
+            }
+
+            memset(buffer, 0, sizeof(void *) * capacity);
+            array->buffer = buffer;
+        }
+    } else {
+        array = malloc(sizeof(struct GrowingArray));
+        array->buffer = calloc(capacity, sizeof(void *));
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)